app/models/User.php

```diff
@@ -0,0 +1,100 @@
+<?php
+namespace app\models;
+
+use app\core\Database;
+use app\interfaces\EntityInterface;
+
+class User implements EntityInterface
+{
+    protected Database $db;
+
+    public function __construct() {
+        $this->db = new Database();
+    }
+    /** Model Create a new user and insert into the database
+     * @param array $data
+     * @return void
+     */
+    public function create(array $data): void
+    {
+        // Hash the password before storing
+        $hashedPassword = password_hash($data['password'], PASSWORD_DEFAULT);
+        $this->db->query("INSERT INTO Users (username, email, password) VALUES (:username, :email, :password)");
+        $this->db->bind(':username', $data['username']);
+        $this->db->bind(':email', $data['email']);
+        $this->db->bind(':password', $hashedPassword);
+        $this->db->execute();
+    }
+
+    /** Model Retrieve a user by username
+     * @param $username
+     * @return mixed
+     */
+    public function findUserByUsername($username)
```

Comment:
this class uses the naming convention `findUserBy` while other models use the style `getPost` - I'd like to see a consistent approach here